src/core/scalebar/qgsdoubleboxscalebarrenderer.cpp

```diff
@@ -190,3 +191,24 @@ void QgsDoubleBoxScaleBarRenderer::draw( QgsRenderContext &context, const QgsSca
   //draw labels using the default method
   drawDefaultLabels( context, settings, scaleContext );
 }
+
+bool QgsDoubleBoxScaleBarRenderer::applyDefaultSettings( QgsScaleBarSettings &settings ) const
+{
+  // restore the fill symbols by default
+  std::unique_ptr< QgsFillSymbol > fillSymbol = std::make_unique< QgsFillSymbol >();
```

Comment:
I'd make this conditional on the settings.fillSymbol() being invisible -- otherwise changing between single box and double box will keep resetting the fill symbol back to default, when we should be able to retain the symbol between these renderers.